internal/runtime/runtime.go

```diff
@@ -37,8 +38,31 @@ func (gr GolangRuntime) ProcessEvent(edgexcontext *appcontext.Context, envelope
 	edgexcontext.LoggingClient.Debug("Processing Event: " + strconv.Itoa(len(gr.Transforms)) + " Transforms")
 	var event models.Event
 
-	if err := json.Unmarshal([]byte(envelope.Payload), &event); err != nil {
-		edgexcontext.LoggingClient.Error("Expected JSON EdgeX Event: " + err.Error())
+	switch envelope.ContentType {
+	case clients.ContentTypeJSON:
+		if err := json.Unmarshal([]byte(envelope.Payload), &event); err != nil {
+			edgexcontext.LoggingClient.Error("Unable to JSON unmarshal  EdgeX Event: "+err.Error(), clients.CorrelationHeader, envelope.CorrelationID)
+			return nil
+		}
+
+		// Needed for Marking event as handled
+		edgexcontext.EventId = event.ID
+
+	case clients.ContentTypeCBOR:
+		event := models.Event{}
+
+		x := codec.CborHandle{}
+		err := codec.NewDecoderBytes([]byte(envelope.Payload), &x).Decode(&event)
+		if err != nil {
```

Comment:
As an FYI -- The requirement in export-distro is simply to just publish the CBOR encoded payload. Looks like you're taking on the work of actually applying additional functions to the event.

Reply:
For App Functions we are just decoding the CBOR into the Event so the functions can work with it. Is export distro also publishing it as CBOR to Rules Engine? Sending to the Rules Engine is a Fuji feature for App Functions.

Reply:
I guess the answer is "yes" since the RulesEngine is potentially a subscriber. But the RulesEngine won't know what to do with the payload in that format. Will have to see.